typescript/sdk/src/router/HyperlaneRouterDeployer.ts

```diff
@@ -151,12 +159,29 @@ export abstract class HyperlaneRouterDeployer<
   async deploy(
     configMap: ChainMap<Config>,
   ): Promise<HyperlaneContractsMap<Factories>> {
-    const contractsMap = await super.deploy(configMap);
+    // Only deploy on chains that don't have existing deployments.
+    const configMapToDeploy = objFilter(
+      configMap,
+      (_chainName, config): config is Config => !config.existingDeployment,
+    );
+    const existingDeployments: ChainMap<types.Address> = objMap(
+      objFilter(
+        configMap,
+        (_chainName, config): config is Config => !!config.existingDeployment,
+      ),
+      (_, config) => config.existingDeployment!,
+    );
```

Comment:
this is hard to read

Reply:
flipped the map & filter and added some comments, hopefully slightly more clear